### ISSUE TITLE
Understanding the higher and non-higher code snippets

### ISSUE THREAD

**Issue by kgarg8** [facebookresearch/higher#80]
Hi, I am implementing some repository which uses a HIGHER code in the below minimalistic snippet.
Since the code uses only parameters at ```time=-1```, I am trying to remove the inner loop_ctx in my non-higher minimalistic snippet.
It looks very clean implementation but I checked that the norm of ```params``` differ a lot which should not be the case in my estimation. In addition, the higher code works but not the non-higher code (the accuracy hardly changes). Do you see any obvious flaw or this looks fine, maybe I am messing up somewhere else.

```
# HIGHER CODE
def inner_loop(self, fmodel, diffopt, train_input, train_target):
        train_logit = fmodel(train_input)
        inner_loss = F.cross_entropy(train_logit, train_target)
        diffopt.step(inner_loss)

def outer_loop(self, batch, is_train):
    for (train_input, train_target, test_input, test_target) in zip(train_inputs, train_targets, test_inputs, test_targets):

        with higher.innerloop_ctx(self.network, self.inner_optimizer, track_higher_grads=False) as (fmodel, diffopt):

            for step in range(self.args.n_inner):
                self.inner_loop(fmodel, diffopt, train_input, train_target)

            if is_train:
                params = list(fmodel.parameters(time=-1))
                in_grad = torch.nn.utils.parameters_to_vector(torch.autograd.grad(in_loss, params, create_graph=True))
                outer_grad = torch.nn.utils.parameters_to_vector(torch.autograd.grad(outer_loss, params))
    if is_train:
        # outer_optimizer.step
```

```
# NON-HIGHER CODE
def new_inner_loop(self, model, opt, train_input, train_target):
        train_logit = model(train_input)
        inner_loss = F.cross_entropy(train_logit, train_target)
        inner_loss.backward()
        opt.step()

def outer_loop(self, batch, is_train):
    for (train_input, train_target, test_input, test_target) in zip(train_inputs, train_targets, test_inputs, test_targets):

        model = self.network
        opt = self.inner_optimizer

        for step in range(self.args.n_inner):
            self.new_inner_loop(model, opt, train_input, train_target) # see the difference

        if is_train:
            params = list(model.parameters()) # see the difference
            # compute in_grad, out_grad
    if is_train:
        # outer_optimizer.step()
````
Thanks
